tt_metal/impl/device/device_pool.hpp

```diff
@@ -0,0 +1,70 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "tt_metal/impl/device/device.hpp"
+#include "impl/debug/dprint_server.hpp"
+#include "impl/debug/watcher_server.hpp"
+#include "tt_metal/third_party/umd/device/tt_cluster_descriptor.h"
+namespace tt {
+
+using Device = tt_metal::Device;
+class DevicePool {
+   public:
+    DevicePool &operator=(const DevicePool &) = delete;
+    DevicePool &operator=(DevicePool &&other) noexcept = delete;
+    DevicePool(const DevicePool &) = delete;
+    DevicePool(DevicePool &&other) noexcept = delete;
+
+
+		static DevicePool& instance() noexcept {
+        TT_ASSERT(_inst != nullptr, "Trying to get DevicePool without initializing it");
+        return *_inst;
+    }
+
+    static void initialize(
+        std::vector<chip_id_t> device_ids,
+        const uint8_t num_hw_cqs,
+        size_t l1_small_size,
+        const std::vector<uint32_t> &l1_bank_remap = {}) noexcept {
+      std::cout << " DP initialize " << std::endl;
+      std::cout << " device ids ";
+      for (const auto &chip: device_ids) { std::cout << chip << " " << std::endl; }
+      std::cout << " num hw cqs " << num_hw_cqs << " l1 small " << l1_small_size << std::endl;
```

Comment:
will remove prints

Reply:
thanks, pls re-ping for approval once done